### ISSUE TITLE
dualsense support in 1.12.2

### ISSUE THREAD

**Issue by Etaash-mathamsetty** [MrCrayfish/Controllable#283]
my dualsense doesn't work in 1.12.2 (mappings are wrong), is it possible to add support for it?

**Comment by dude98** [comment]
use something like DS4Windows for now to emulate an xbox controller while it's not supported, as he's been releasing the mod for newer versions of the game and might not go back to that version to update it.

**Comment by Etaash-mathamsetty** [comment]
> use something like DS4Windows for now to emulate an xbox controller while it's not supported, as he's been releasing the mod for newer versions of the game and might not go back to that version to update it.

lmao windows, aint no way im ever using that again
but actually, its kinda annoying to run ds360 every single time, when he can just backport the patch rather easily

**Comment by dude98** [comment]
i don't see why you have an issue with ds4windows, it works quite well to me. (or do you mean windows windows? if that's the case, then i agree with windows being weird some days, but ds4windows is a controller remapper, not what you think it is.)

if you want a possible idea though, and you know how to code, if he's ok with it, you can take the 1.18 gamecontrollerdb.txt file, put it in the 1.12.2 jar file, and launch the game with the new controller set and see if it sees your controller, as each controller has to have a spot in that file or it won't be seen. if it still doesn't work, then something else in the files have changed and it could be anything.

**Comment by Etaash-mathamsetty** [comment]
> i don't see why you have an issue with ds4windows, it works quite well to me. (or do you mean windows windows? if that's the case, then i agree with windows being weird some days, but ds4windows is a controller remapper, not what you think it is.)
> 
> if you want a possible idea though, and you know how to code, if he's ok with it, you can take the 1.18 gamecontrollerdb.txt file, put it in the 1.12.2 jar file, and launch the game with the new controller set and see if it sees your controller, as each controller has to have a spot in that file or it won't be seen. if it still doesn't work, then something else in the files have changed and it could be anything.

I don't think you realized that I don't use windows
but yeah that's a great solution and the one that I was looking for
aaaand it didn't work

**Comment by dude98** [comment]
nothing you wrote implied you dont use windows, and 90 percent of people do.
and i said possible idea. if it didn't work then something in the program is different. you might have to backport it yourself since he's busy working on other things and a controller support mod for an older version of minecraft isn't exactly high priority.

**Comment by MrCrayfish** [comment]
Any other platform besides Windows is unfortuantely going to be questionable when it comes to controller support. Controllable only provides the correct mappings so if your controller does work, it will be mapped correctly in the game. This is why tools like DS4Windows exist since it extends the list of supported games for your controller. It might have changed but Minecraft Dungeons didn't support the PS4 Controller when I played it so I had to use DS4Windows to emulate an Xbox controller.

Having that said, it may be as simple as updating the mapping database since the 1.12.2 version of the mod was released before the PS5 was even annouced (how would I add mappings for controller that doesn't exist?). You can see the potential solution here: https://github.com/MrCrayfish/Controllable/issues/268#issuecomment-1099176826

**Comment by Etaash-mathamsetty** [comment]
> Any other platform besides Windows is unfortuantely going to be questionable when it comes to controller support. Controllable only provides the correct mappings so if your controller does work, it will be mapped correctly in the game. This is why tools like DS4Windows exist since it extends the list of supported games for your controller. It might have changed but Minecraft Dungeons didn't support the PS4 Controller when I played it so I had to use DS4Windows to emulate an Xbox controller.
> 
> Having that said, it may be as simple as updating the mapping database since the 1.12.2 version of the mod was released before the PS5 was even annouced (how would I add mappings for controller that doesn't exist?). You can see the potential solution here: [#268 (comment)](https://github.com/MrCrayfish/Controllable/issues/268#issuecomment-1099176826)

that solution didn't work, like I said earlier
let me try again though, maybe the flatpak sandbox is screwing up things

**Comment by Etaash-mathamsetty** [comment]
mapping is present:
```
Controller,a:b1,b:b2,back:b8,dpdown:h0.4,dpleft:h0.8,dpright:h0.2,dpup:h0.1,guide:b12,leftshoulder:b4,leftstick:b10,lefttrigger:a3,leftx:a0,lefty:a1,rightshoulder:b5,rightstick:b11,righttrigger:a4,rightx:a2,righty:a5,start:b9,touchpad:b13,x:b0,y:b3,platform:Linux,
050000004c050000e60c000000010000,PS5
```
its not flatpak, its broken in minecraft launcher too
well guess I got to run a script every time I want to play minecraft with a controller now :/

**Comment by dude98** [comment]
okay so that line you took from the controller database file looks backwards or you accidentally took the top half of the first and bottom half of the next. it needs to be so that the platform is at the end of the line with the controller code and name at the start of the line.
also, there are 2 separate ps5 controller IDs, you need to put both in, in case you put the one with the wrong ID in.

**Comment by Etaash-mathamsetty** [comment]
> okay so that line you took from the controller database file looks backwards or you accidentally took the top half of the first and bottom half of the next. it needs to be so that the platform is at the end of the line with the controller code and name at the start of the line. also, there are 2 separate ps5 controller IDs, you need to put both in, in case you put the one with the wrong ID in.

yeah I know, I just copied the whole file lol, this is just an excerpt of the file in the jar file

**Comment by MrCrayfish** [comment]
I've written some guides and troubleshooting information for Controllable. Maybe something there will help solve your issue, otherwise I no longer maintain the 1.12.2 version of this mod.
https://mrcrayfish.gitbook.io/controllable-documentation/
